test/kfp-functional-test/kfp-functional-test.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash -ex
+# Copyright 2020 Kubeflow Pipelines contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if [[ ! -z "${GOOGLE_APPLICATION_CREDENTIALS}" ]]; then
+  # activating the service account
+  gcloud auth activate-service-account --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
+fi
+source_root=$(pwd)
+
+# TODO(#4853) Skipping pip 20.3 due to a bad version resolution logic.
+python3 -m pip install --upgrade pip!=20.3.*
+python3 -m pip install -r "$source_root/test/kfp-functional-test/requirements.txt"
+
+TEST_RESULT_BUCKET=ml-pipeline-test
+TEST_RESULT_FOLDER=kfp-functional-e2e-test
+TEST_RESULTS_GCS_DIR=gs://${TEST_RESULT_BUCKET}/${TEST_RESULT_FOLDER}
```

Comment:
FYI, when other tests use the result GCS DIR, it was because the test script runs on prow pod (like your script), but the python test script runs in KFP cluster. So it was necessary to store temporary test results in a GCS bucket and export it to prow at the end in the prow pod like in https://github.com/kubeflow/pipelines/blob/04d5e609f224d7be2e8bfc20a3fa7857906cae1c/test/check-argo-status.sh#L72.

However, in your case, the your python test script runs in the prow pod, so it can directly dump all results to a local path `${ARTIFACTS}` and prow will pick it up.
For prow's documentation of this behavior, read https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#job-environment-variables